app/services/products_renderer.rb

```diff
@@ -0,0 +1,92 @@
+require 'open_food_network/scope_product_to_hub'
+
+class ProductsRenderer
+  class NoProducts < RuntimeError; end
+
+  def initialize(distributor, order_cycle, customer, args = {})
+    @distributor = distributor
+    @order_cycle = order_cycle
+    @customer = customer
+    @args = args
+  end
+
+  def products_json
+    raise NoProducts unless order_cycle && distributor && products
+
+    ActiveModel::ArraySerializer.new(products,
+                                     each_serializer: Api::ProductSerializer,
+                                     current_order_cycle: order_cycle,
+                                     current_distributor: distributor,
+                                     variants: variants_for_shop_by_id,
+                                     master_variants: master_variants_for_shop_by_id,
+                                     enterprise_fee_calculator: enterprise_fee_calculator).to_json
+  end
+
+  private
+
+  attr_reader :order_cycle, :distributor, :customer, :args
+
+  def products
+    return unless order_cycle
+
+    @products ||= begin
+      results = distributed_products.products_relation.order(taxon_order)
+
+      filter_and_paginate(results).
+        each { |product| product_scoper.scope(product) } # Scope results with variant_overrides
+    end
+  end
+
+  def product_scoper
+    OpenFoodNetwork::ScopeProductToHub.new(distributor)
+  end
+
+  def enterprise_fee_calculator
+    OpenFoodNetwork::EnterpriseFeeCalculator.new distributor, order_cycle
+  end
+
+  def filter_and_paginate(query)
+    query.ransack(args[:q]).result.page(args[:page] || 1).per(args[:per_page] || 10)
+  end
+
+  def distributed_products
+    OrderCycleDistributedProducts.new(distributor, order_cycle, customer)
+  end
+
+  def taxon_order
+    if distributor.preferred_shopfront_taxon_order.present?
+      distributor
+        .preferred_shopfront_taxon_order
+        .split(",").map { |id| "primary_taxon_id=#{id} DESC" }
```

Comment:
Again, I'd be consistent and have all lines separate.

Reply:
I agree in general but Matt just moved this code and didn't write it. The pull request is big enough as is and we should separate concerns.